src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -208,6 +210,10 @@ def get_build_options(
                 error_cls=DockerImageOptionValueError,
             )
             yield from target[field_type].options(format)
+        elif issubclass(field_type, DockerBuildUnaryOptionFieldMixin):
+            yield from target[field_type].options()
+        elif issubclass(field_type, DockerBuildSkippingOptionFieldMixin):
+            yield from target[field_type].options()
```

Comment:
Addressing my comments regarding the field mixin classes, I hope these two new `elif` branches can go away again.

If the extra instantiation of the interpolation context is a concern, that could be refactored into a classmethod on the field mixin instead, and only instantiated for those that actually use it.

Reply:
I see. I'll leave it to the follow-up PR. I got quite a few other flags I'd like to configure (e.g. `--platform`)